src/components/NavBar/NavBar.tsx

```diff
@@ -1,37 +1,85 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 import { breakpoints } from '../../styles/breakpoints';
 import { colours } from '../../styles/colours';
 import { Typography } from '../Typography/Typography';
 
+/**
+ * Interface for the NavBarProps.
+ * @property {boolean} isVisible - Indicates whether the navigation bar is visible or not.
+ */
```

Comment:
This code here is pretty clear. There's probably no need for these comments here, unless these are used for auto generated docs?

Reply:
Some comments are used for docs. In general, I commented more than usual as we were considering bringing on interns to the frontend and most members on the team were more familiar with backend code than the frontend.